src/main/java/io/cryostat/net/TargetConnectionManager.java

```diff
@@ -90,7 +90,8 @@
                                     public void onRemoval(
                                             ConnectionDescriptor descriptor,
                                             JFRConnection connection,
-                                            RemovalCause cause) {
+                                            RemovalCause cause)
+                                            throws RuntimeException {
```

Comment:
I don't think this line is required, is it? You shouldn't need to declare that you throw unchecked exceptions.

Reply:
You're right, I'll go head and change that.